src/raw.rs

```diff
@@ -179,6 +188,35 @@ impl RawValue {
         Ok(Self::from_owned(json.into_boxed_str()))
     }
 
+    /// Convert an owned `String` of JSON data to an owned `RawValue`, without
+    /// checking whether the JSON is valid.
+    ///
+    /// We avoid an allocation and memcpy if the input has capacity equal to
+    /// its length.
+    ///
+    /// # Safety
+    ///
+    /// This function assumes that:
+    ///
+    /// - the input is valid JSON,
+    /// - the input has no leading or trailing whitespace.
+    pub unsafe fn from_string_unchecked(json: String) -> Box<Self> {
+        Self::from_owned(json.into_boxed_str())
+    }
+
+    /// Convert a borrowed `str` of JSON data to a borrowed `RawValue`, without
+    /// checking whether the JSON is valid.
+    ///
+    /// # Safety
+    ///
+    /// This function assumes that:
+    ///
+    /// - the input is valid JSON, and
+    /// - the input has no leading or trailing whitespace.
+    pub unsafe fn from_str_unchecked(json: &str) -> &Self {
+        Self::from_borrowed(json)
+    }
```

Comment:
I would like to avoid adding these functions. Do you have a real use case where these are essential?